middleware/src/middleware.go

```diff
@@ -522,37 +668,100 @@ func (middleware *Middleware) SetRootPassword(args rpcmessages.SetRootPasswordAr
 	// len([]rune("₿")) = 1
 	if len([]rune(password)) >= 8 {
 		out, err := middleware.runBBBConfigScript([]string{"set", "root_pw", password})
+
 		if err != nil {
-			return rpcmessages.ErrorResponse{Success: false, Message: string(out), Code: err.Error()}
+			errorCode := handleBBBScriptError(out, err, []rpcmessages.ErrorCode{
+				rpcmessages.ErrorSetNeedsTwoArguments,
+			})
+
+			return rpcmessages.ErrorResponse{
+				Success: false,
+				Message: strings.Join(out, "\n"),
+				Code:    errorCode,
+			}
 		}
+
 		return rpcmessages.ErrorResponse{Success: true}
 	}
-	return rpcmessages.ErrorResponse{Success: false, Message: "invalid password"}
+
+	return rpcmessages.ErrorResponse{
+		Success: false,
+		Message: "The password has to be at least 8 chars. An unicode char is counted as one.",
+		Code:    rpcmessages.ErrorSetRootPasswordToShort,
+	}
 }
 
 // runBBBCmdScript runs the bbb-cmd.sh script.
 // The script executes commands like for example mounting a USB drive, doing a backup and copying files.
-func (middleware *Middleware) runBBBCmdScript(args []string) (out []byte, err error) {
-	script := middleware.environment.GetBBBCmdScript()
-	cmdAsString := script + " " + strings.Join(args, " ")
-	out, err = exec.Command(script, args...).Output()
-	if err != nil {
-		// no error handling here, only logging.
-		log.Printf("Error: The command '%s' exited with the output '%v' and error '%s'.\n", cmdAsString, string(out), err.Error())
-	}
+func (middleware *Middleware) runBBBCmdScript(args []string) (outputLines []string, err error) {
+	outputLines, err = runCommand(middleware.environment.GetBBBCmdScript(), args)
 	return
 }
 
 // runBBBConfigScript runs the bbb-config.sh script.
 // The script changes the system configuration in redis by setting or unsetting the appropriate keys.
 // If necessary the affected services are restarted.
-func (middleware *Middleware) runBBBConfigScript(args []string) (out []byte, err error) {
-	script := middleware.environment.GetBBBConfigScript()
-	cmdAsString := script + " " + strings.Join(args, " ")
-	out, err = exec.Command(script, args...).Output()
+func (middleware *Middleware) runBBBConfigScript(args []string) (outputLines []string, err error) {
+	outputLines, err = runCommand(middleware.environment.GetBBBConfigScript(), args)
+	return
+}
+
+// runCommand runs the passed command and returns the combined stdout and stderr output.
+// If the command could not be run, err is not nil.
+func runCommand(command string, args []string) (combinedLines []string, err error) {
```

Comment:
What about typing the `command string` parameter instead with something like `type BBBScript` which could be either `BBBCmdScript` or `BBBConfigScript` so `runCommand` only accepts predefined scripts instead of any string?

Reply:
To replace `runBBBCmdScript()` and `runBBBConfigScript()` ?

Reply:
Sorry, should have been clearer, what I meant was something like this:
`func runCommand(command BBBscript, args []string) (combinedLines []string, err error) {...}`
but then I suppose we would need `GetBBBConfigScript()` to return `type BBBScript string` first anyway. My thought was to just make the parameters `runCommand()` accepts more specific. But let's just say that's out of scope for this PR. Something to keep in mind for future refactoring